R/MsBackend.R

```diff
@@ -849,6 +861,13 @@ setMethod("backendMerge", "MsBackend", function(object, ...) {
     stop("Not implemented for ", class(object), ".")
 })
 
+#' @exportMethod backendParallelFactor
+#'
+#' @rdname MsBackend
+setMethod("backendParallelFactor", "MsBackend", function(object, ...) {
+    factor()
```

Comment:
Trying to visualize what happen between different uses, but essentially why do we need this method ? Because it will only be used by users that have datasets with `MsBackendHdf5Peaks` or `MsBackendMzR` no ? Or is it just mandatory to have a default implementation ? you cannot just leave the function throwing an error if user try to use this function with other backends ?

Reply:
actually yes, why isn't it define exactly like the method below for `export` ?

Reply:
the idea is that with this method we allow the backend to suggest an (optimal) factor to split the data. Here we define a default implementation for **any** `MsBackend` class, that simply returns `factor()`, i.e. does not suggest any splitting.  For `MsBackendMzR` and `MsBackendHdf5Peaks` we overwrite this method to actually suggest a splitting by original data file (or rather data storage). 

We will call `backendParallelFactor(object@backend)` (`object` being a `Spectra`) for some data access or processing methods to evaluate whether we need to split the data for chunk-wise processing. If we would throw an error like below every `MsBackend` implementation would need to implement this method (even if it will return `factor()` like here with the default implementation). Having an actual default implementation here reduces the code that needs to be implemented for each backend (and also reduces code repetition, because all backends for now, except the two mentioned above) will not provide any suggested splitting.